projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -1,24 +1,287 @@
+      }
+    } catch (IOException e) {
+    }
+    return allFaresAttributesBuilder;
+  }
+
+  public static FaresRules.Builder getFareRules(Path csvPath) {
+    FaresRules.Builder allFaresRulesBuilder = FaresRules.newBuilder();
+    try {
+      CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT);
+      List<CSVRecord> csvRecords = parser.getRecords();
+      // remove the first record which are columns title
+      csvRecords.remove(csvRecords.get(0));
+      // iterate each record, create a message with builder
+      for (CSVRecord record : csvRecords) {
+        allFaresRulesBuilder.addFaresrules(FareRules.newBuilder().setFareId(Integer.parseInt(record.get(0)))
+            .setRouteId(Integer.parseInt(record.get(1))).setOriginId(Integer.parseInt(record.get(2)))
+            .setDestinationId(Integer.parseInt(record.get(3))).setContainsId(record.get(4)).build());
+      }
+    } catch (IOException e) {
+    }
+    return allFaresRulesBuilder;
+  }
+
+  public static Routes.Builder getRoutes(Path csvPath) {
+    Routes.Builder allRoutesBuilder = Routes.newBuilder();
+    try {
+      CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT);
+      List<CSVRecord> csvRecords = parser.getRecords();
+
+      // remove the first record which are columns title
+      csvRecords.remove(csvRecords.get(0));
+      // iterate each record, create a message with builder
+      for (CSVRecord record : csvRecords) {
+        allRoutesBuilder.addRoutes(
+            Route.newBuilder().setRouteId(Integer.parseInt(record.get(0))).setAgencyId(Integer.parseInt(record.get(1)))
+                .setRouteShortName(record.get(2)).setRouteLongName(record.get(3)).setRouteDesc(record.get(4))
+                .setRouteType(Integer.parseInt(record.get(5))).setRouteColor(record.get(6)).build());
+      }
+    } catch (IOException e) {
+    }
+    return allRoutesBuilder;
+  }
+
+  public static Shapes.Builder getShapes(Path csvPath) {
+    Shapes.Builder allShapesBuilder = Shapes.newBuilder();
+    try {
+      CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT);
+      List<CSVRecord> csvRecords = parser.getRecords();
+
+      // remove the first record which are columns title
+      csvRecords.remove(csvRecords.get(0));
+      // iterate each record, create a message with builder
+      for (CSVRecord record : csvRecords) {
+        allShapesBuilder.addShapes(Shape.newBuilder().setShapeId(Integer.parseInt(record.get(0)))
+            .setShapePtLat(Double.parseDouble(record.get(1))).setShapePtLot(Double.parseDouble(record.get(2)))
+            .setShapePtSequence(Integer.parseInt(record.get(3))).build());
+      }
+    } catch (IOException e) {
+    }
+    return allShapesBuilder;
+  }
+
+  public static void getStopTime(Path csvPath, String path) {
+    try {
+      //where we serialize the messages
+      FileOutputStream output = new FileOutputStream(path);

```

Comment:
Looks like you don't close the stream. You can use try-with-resources construction.

Reply:
Can you rephrase, please?